studio/pages/project/[ref]/settings/logs/[type].tsx

```diff
@@ -0,0 +1,109 @@
+import debounce from 'lodash/debounce'
+import CodeEditor from 'components/ui/CodeEditor'
+
+interface CountData {
+  count: number
+}
+/**
+ * Acts as a container component for the entire log display
+ */
+export const LogPage = () => {
+  const router = useRouter()
+  const { ref, type } = router.query
+
+  const [search, setSearch] = useState<string>('')
+  const [queryParams, setQueryParams] = useState<string>('')
+  const [where, setWhere] = useState<string>('')
+  const [mode, setMode] = useState<'simple' | 'custom'>('simple')
+  const [latestRefresh, setLatestRefresh] = useState<string>(new Date().toISOString())
+
+  const title = LOG_TYPE_LABEL_MAPPING[type as string]
+
+  // handle log fetching
+  const {
+    data,
+    isValidating: isLoading,
+    mutate,
+  } = useSWR<{ data: LogData[]; count: number; error: any }>(
+    `${API_URL}/projects/${ref}/logs?${queryParams}`,
+    get,
+    {
+      revalidateOnFocus: false,
+    }
+  )
+
+  const debouncedQueryParams = useRef(debounce(setQueryParams, 600)).current
+
+  useEffect(() => {
+    const params = {
+      type: type as string,
+      search_query: search || '',
+      where: where || '',
+    }
+    const qs = new URLSearchParams(params).toString()
+    debouncedQueryParams(qs)
+    return () => debouncedQueryParams.cancel()
+  }, [search, where, type])
+
+  const countKey = `${API_URL}/projects/${ref}/logs?${queryParams}&count=true&period_start=${latestRefresh}`
+  const { data: countData } = useSWR<{ data: [CountData] | []; error?: any }>(countKey, get, {
+    refreshInterval: 5000,
+  })
+  const newCount = countData?.data?.[0]?.count
+  const { data: logData, error } = data || {}
+
+  const handleRefresh = () => {
+    setLatestRefresh(new Date().toISOString())
+    mutate()
+  }
+  const handleModeToggle = () => {
+    if (mode === 'simple') {
+      setMode('custom')
+      setSearch('')
+    } else {
+      setMode('simple')
+    }
+  }
+  return (
+    <SettingsLayout title={title} className="p-4 space-y-4">
+      <LogPanel
+        presets={[
+          { label: "Recent - All", onClick: () => setSearch('') },
+          { label: "Recent - Errors", onClick: () => setSearch('[Ee]rror|\\s[45][0-9][0-9]\\s') },
+        ]}
+        searchValue={search}
+        onCustomClick={handleModeToggle} isLoading={isLoading} onRefresh={handleRefresh} onSearch={setSearch} />
+      {mode === 'custom' &&
+        <div>
+          <CodeEditor
+            className="p-4 h-24"
+            hideLineNumbers
+            id={'logs-where-editor'}
+            language="pgsql"
+            defaultValue={where}
+            onInputChange={(v) => setWhere(v || '')}
+            onInputRun={handleRefresh}
+          />
+        </div>}
+      {error && (
+        <Typography.Text className="text-center w-full block">Could not fetch data</Typography.Text>
+      )}
+      {newCount ? <Button type="dashed" onClick={handleRefresh} block>Load new logs</Button> : null}
```

Comment:
extract this button to a component. cleaner code.

Reply:
Noted, will defer the swr data refactoring to when i'll be adding in the 'Load older logs' button, as logic will have to be adjusted anyway.